docs/FLUTTER_STYLE.md

```diff
@@ -0,0 +1,137 @@
+# Code Style
+
+All rules fall through to the [Effective Dart](https://dart.dev/guides/language/effective-dart)
+guidelines.
+
+## 1. All comments, identifiers, issues, etc. should be written in English
+
+## 2. Limit code to contain 80 characters per line
+
+Exceptions:
+
+- documentation code
+- hard-coded URLs (e.g., in comments)
+- TODO comments should stay on a single line, so that tools can read & report
+  the full message
+
+## 3. Single- vs. Multi-line function calls
+
+Put calls in a single line if they fit:
+
+```dart
+return Container(color: Colors.pink, child: myChild);
+```
+
+Wrap if there are more than two named parameters, they get longer:
+
+```dart
+return Container(
+  color: context.theme.primaryColor,
+  padding: EdgeInsets.symmetric(horizontal: 16),
+  child: myChild,
+);
+```
+
+## 4. Always prefer relative to absolute imports
+
+E.g. `import '../common/module.dart'` instead of `import '/common/module.dart'`
+
+## 5. Use private functions/classes if they are only used locally
+
+In Flutter, you make symbols private by prefixing them with an underscore (e.g. `_DatePickerState`).
+
+## 6. Grouping and ordering concepts within a file
+
+- Group concepts by their functionality vs. e.g. their return type
+- Order these groups of concepts within a file by their relevance
+
+  - E.g. Put helper methods beneath the methods they help
+
+    Why? The name of helper methods should convey their function so you don't
+    need to view their contents to get an initial understanding of the class.
```

Comment:
I don't really understand this reasoning: I don't understand why the helper methods should be beneath the methods they are helping because they have a meaningful name

Reply:
Say we have a page in the app and there are a few widgets that are specific to this page within our app, being used in no other place.
Then it would make sense to put them in the file of the page and make them private to the scope of the file.

In this case, the two specific widgets are 'helper concepts'. In an OOP context, i would argue it is more intuitive that when you open the page file, the first concept you are presented with is the page itself. Especially when the project is in maintenance, this allows for developers only to dive as deep into the code as they need to.

Of course this rule is arguable but either way i would like to establish a rule concerned with the structure of a file just to further unify our code base. If you would prefer putting the helper methods first, feel free to put a suggestion in a code review on this PR or we can talk about it in the daily tomorrow :)

Reply:
Thanks for explaining, although that's not what I meant. My point was that I don't understand why the helper methods should be beneath the methods they are helping **because** they have a meaningful name. In my mind, these are two different points. Shouldn't they have a meaningful name no matter where they are placed? 
We can go with this rule as suggested.

Reply:
Alright, I'll re-request a review from you

Reply:
Somehow it's not letting me request a review again ¯\_(ツ)_/¯

Can you submit another review yourself? @dieknolle3333